### PR TITLE
training on different traffic histories

### PR DESCRIPTION
**Pull request by YuechengLiu**

### CODE REVIEW

**Comment by zbzhu99** [comment]
@YuechengLiu great PR!
By the way, please take a look at all the new alerts introduced by this PR mentioned by LGTM :smile_cat:

**Comment by YuechengLiu** [comment]
> @YuechengLiu great PR! By the way, please take a look at all the new alerts introduced by this PR mentioned by LGTM 😸

I will fix them soon.

**Comment by YuechengLiu** [comment]
> @YuechengLiu great PR! By the way, please take a look at all the new alerts introduced by this PR mentioned by LGTM 😸

I tried to fix these errors, and I fixed some of them. But it seems that some of the run_scripts are not compatible with `rlkit.envs.get_envs`, even before my last commit.

**Comment by zbzhu99** [comment]
> 

Sure, some of the run_scripts may be outdated for some time. I will check them before merging.

**Comment by zbzhu99** [comment]
> > @YuechengLiu great PR! By the way, please take a look at all the new alerts introduced by this PR mentioned by LGTM 😸
> 
> I tried to fix these errors, and I fixed some of them. But it seems that some of the run_scripts are not compatible with `rlkit.envs.get_envs`, even before my last commit.

Sure, some of the run_scripts may be outdated for some time. I will check them before merging.

**Comment by zbzhu99** [comment]
I will merge this PR today.

File: ILSwiss/rlkit/envs/__init__.py [hunk 1]
```diff
@@ -51,31 +52,20 @@ def get_envs(
...
-        envs = DummyVectorEnv(
-            [
-                lambda i=i: env_wrapper(
-                    env_class(vehicle_ids=vehicle_ids_list[i], **env_specs)
-                )
-                for i in range(env_num)
-            ],
-            auto_reset=auto_reset,
-            **kwargs,
-        )
```
Comment (zbzhu99) [inline ILSwiss/rlkit/envs/__init__.py]: DummyVectorEnv should be kept for debugging purposes. In some cases, debugging something that happens in another subprocess is troublesome.

File: ILSwiss/rlkit/samplers/__init__.py [hunk 2]
```diff
@@ -59,6 +59,7 @@ def obtain_samples(self, num_steps=None, pred_obs=False):
 
         finished_env_ids = []
         env_finished_car_num = np.zeros(self.env_num)
+
```
Comment (zbzhu99) [inline ILSwiss/rlkit/samplers/__init__.py]: I will restore this change

**Comment by zbzhu99** [comment]
Thank you for fixing problems in run_scripts of other algorithms :rose:

File: ILSwiss/run_scripts/gen_smarts_demos.py [hunk 3]
```diff
@@ -194,6 +223,58 @@ def sample_demos(
...
+    demo_trajs = []
+    while True:  # not reliable
+        try:
+            if len(demo_trajs) == 0:
+                traj = trajs_queue.get(block=True)
+            else:
+                traj = trajs_queue.get(block=True, timeout=100)
+            demo_trajs.append(traj)
+            print(f"main process: collected trajs num: {len(demo_trajs)}")
+        except:
```
Comment (zbzhu99) [inline ILSwiss/run_scripts/gen_smarts_demos.py]: LSTM suggests not using bare except to catch all exceptions, which may hide some other problems.
I will change this to only catch `queue.Empty` exception.

**Pull request merged**